src/Microsoft.Data/DataFrame.cs

```diff
@@ -33,6 +36,273 @@ public DataFrame(IList<BaseColumn> columns)
+                        }
+                        break;
+                    case ArrowTypeId.Binary:
+                        ReadOnlyMemory<byte> byteValueBuffer = (arrowArray as PrimitiveArray<byte>).ValueBuffer.Memory;
+                        ReadOnlyMemory<byte> byteNullBitMapBuffer = (arrowArray as PrimitiveArray<byte>).NullBitmapBuffer.Memory;
+                        dataFrameMemory = new Memory<byte>(new byte[byteValueBuffer.Length]);
+                        nullBitMapMemory = new Memory<byte>(new byte[byteNullBitMapBuffer.Length]);
+                        copied = byteValueBuffer.TryCopyTo(dataFrameMemory) && byteNullBitMapBuffer.TryCopyTo(nullBitMapMemory);
+                        if (copied)
+                        {
+                            dataFrameColumn = new PrimitiveColumn<byte>(field.Name, dataFrameMemory, nullBitMapMemory, arrowArray.Length);
+                        }
+                        break;
+                    case ArrowTypeId.Date32:
+                    case ArrowTypeId.Date64:
+                    case ArrowTypeId.Dictionary:
+                    case ArrowTypeId.FixedSizedBinary:
+                    case ArrowTypeId.HalfFloat:
+                    case ArrowTypeId.Interval:
+                    case ArrowTypeId.List:
+                    case ArrowTypeId.Map:
+                    case ArrowTypeId.Null:
+                    case ArrowTypeId.Struct:
+                    case ArrowTypeId.Time32:
+                    case ArrowTypeId.Time64:
+                    case ArrowTypeId.Timestamp:
+                    case ArrowTypeId.Union:
+                    default:
+                        throw new NotImplementedException(nameof(fieldType.Name));
+                }
+                _table.InsertColumn(ColumnCount, dataFrameColumn);
+                fieldIndex++;
+            }
+        }
+
+        public IEnumerable<RecordBatch> AsArrowRecordBatch()
+        {
+            Apache.Arrow.Schema.Builder schemaBuilder = new Apache.Arrow.Schema.Builder();
+            Field.Builder fieldBuilder = new Field.Builder();
+            // TODO: Sanity check. Some column types are not Arrow Compatible
+            // Derived column types will NOT be supported
+
+            // Find the number of DataFrameBuffers in each column. All columns have the same number of buffers
+            int numberOfColumns = ColumnCount;
+            // Assuming that all columns are Arrow compatible 
+            int numberOfRecordBatches = 0;
+            for (int i = 0; i < numberOfColumns; i++)
+            {
+                BaseColumn column = Column(i);
+                switch (column)
+                {
+                    case PrimitiveColumn<bool> boolColumn:
+                        numberOfRecordBatches = boolColumn.NumberOfBuffers;
+                        Field boolField = fieldBuilder.Name(boolColumn.Name).Nullable(true).DataType(BooleanType.Default).Build();
```

Comment:
Why are they always `Nullable`? What if this DataFrame doesn't contain any nulls in that column? Should it still be Nullable?

Reply:
It need not be `Nullable`. I was just hacking in this patch and testing Null support in C# Arrow